frontend/src/context/user-prefs-context.tsx

```diff
@@ -17,23 +19,31 @@ const UserPrefsContext = React.createContext<UserPrefsContextType | undefined>(
   undefined,
 );
 
+const SETTINGS_QUERY_KEY = ["settings"];
+
 function UserPrefsProvider({ children }: React.PropsWithChildren) {
-  const [settings, setSettings] = React.useState(getSettings());
+  const { data: settings } = useQuery({
+    queryKey: SETTINGS_QUERY_KEY,
+    queryFn: getSettings,
+    initialData: DEFAULT_SETTINGS,
+  });
+
   const [settingsAreUpToDate, setSettingsAreUpToDate] = React.useState(
     checkIfSettingsAreUpToDate(),
   );
+  const queryClient = useQueryClient();
 
   const saveSettings = (newSettings: Partial<Settings>) => {
     updateAndSaveSettingsToLocalStorage(newSettings);
```

Comment:
This function name should probably change?

Reply:
Done!